addons/l10n_ch/models/ir_actions_report.py

```diff
@@ -27,40 +27,42 @@ def apply_qr_code_ch_cross_mask(self, width, height, barcode_drawing):
         qr_cross = ReportLabImage((width/2 - cross_width/2) / mm, (height/2 - cross_height/2) / mm, cross_width / mm, cross_height / mm, cross_path.as_posix())
         barcode_drawing.add(qr_cross)
 
-    def _render_qweb_pdf_prepare_streams(self, data, res_ids=None):
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # OVERRIDE
-        res = super()._render_qweb_pdf_prepare_streams(data, res_ids)
-        if res_ids and self.report_name in ('account.report_invoice_with_payments', 'account.report_invoice'):
-            invoices = self.env[self.model].browse(res_ids)
-            # Determine which invoices need a QR/ISR.
-            qr_inv_ids = []
-            isr_inv_ids = []
-            for invoice in invoices:
-                if invoice.company_id.country_code != 'CH':
-                    continue
-                if invoice.l10n_ch_is_qr_valid:
-                    qr_inv_ids.append(invoice.id)
-                elif invoice.l10n_ch_isr_valid:
-                    isr_inv_ids.append(invoice.id)
-            # Render the additional reports.
-            streams_to_append = {}
-            if qr_inv_ids:
-                qr_res = self.env.ref('l10n_ch.l10n_ch_qr_report')._render_qweb_pdf_prepare_streams(data, res_ids=qr_inv_ids)
-                for invoice_id, stream in qr_res.items():
-                    streams_to_append[invoice_id] = stream
-            if isr_inv_ids:
-                isr_res = self.env.ref('l10n_ch.l10n_ch_isr_report')._render_qweb_pdf_prepare_streams(data, res_ids=qr_inv_ids)
-                for invoice_id, stream in isr_res.items():
-                    streams_to_append[invoice_id] = stream
-            # Add to results
-            for invoice_id, additional_stream in streams_to_append.items():
-                invoice_stream = res[invoice_id]['stream']
-                writer = OdooPdfFileWriter()
-                writer.appendPagesFromReader(OdooPdfFileReader(invoice_stream, strict=False))
-                writer.appendPagesFromReader(OdooPdfFileReader(additional_stream['stream'], strict=False))
-                new_pdf_stream = io.BytesIO()
-                writer.write(new_pdf_stream)
-                res[invoice_id]['stream'] = new_pdf_stream
-                invoice_stream.close()
-                additional_stream['stream'].close()
+        res = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids)
+        if res_ids:
+            report = self._get_report(report_ref)
+            if report.report_name in ('account.report_invoice_with_payments', 'account.report_invoice'):
```

Comment:
If this is not going to be backported, it might be a legit place to use `:=` in in order to avoid changing the entire indenation of the body? e.g.

```suggestion
        if res_ids and (report := self._get_report(report_ref)).report_name in ('account.report_invoice_with_payments', 'account.report_invoice'):
```